pkg/util/generic.go

```diff
@@ -101,6 +101,20 @@ func FieldIndexByName[T proto.Message](name string) int {
 	return -1
 }
 
+type Collection[T any] interface {
+	Len() int
+	Get(i int) T
+}
+
+func Collect[T any, C Collection[T]](c C) []T {
```

Comment:
`Collect` is a little ambiguous

Reply:
I don't think this was actually used, so I can remove it for now. It's specifically for these guys 
![image](https://github.com/rancher/opni/assets/8194899/cf76a331-40ab-4a85-9aad-9c3fe512e90b)